merge.go

```diff
@@ -46,7 +46,6 @@ func deepMerge(dst, src reflect.Value, visited map[uintptr]*visit, depth int, co
 	typeCheck := config.TypeCheck
 	overwriteWithEmptySrc := config.overwriteWithEmptyValue
 	overwriteSliceWithEmptySrc := config.overwriteSliceWithEmptyValue
-	config.overwriteWithEmptyValue = false
```

Comment:
this is forced true in https://github.com/imdario/mergo/blob/master/map.go#L75

by dropping the reset to false, any merge involving a map leaves this value set to `true` after merging the map.